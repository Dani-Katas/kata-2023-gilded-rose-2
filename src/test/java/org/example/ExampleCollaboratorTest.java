package org.example;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

class ExampleCollaboratorTest {

  @Test
  void sum() {
    CombinationApprovals.verifyAllCombinations(
        this::doSum,
        new Integer[] {-1, 2, 3, 4},
        new Integer[] {-1, 2, 3, 4}
    );
  }

  private int doSum(final Integer a, final Integer b) {
    final ExampleCollaborator exampleCollaborator = new ExampleCollaborator();
    return exampleCollaborator.sum(a, b);
  }
}