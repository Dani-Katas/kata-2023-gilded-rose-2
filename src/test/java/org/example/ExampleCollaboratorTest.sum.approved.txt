[-1, -1] => -2 
[-1, 2] => 1 
[-1, 3] => 2 
[-1, 4] => 3 
[2, -1] => 1 
[2, 2] => 4 
[2, 3] => 5 
[2, 4] => 6 
[3, -1] => 2 
[3, 2] => 5 
[3, 3] => 6 
[3, 4] => 7 
[4, -1] => 3 
[4, 2] => 6 
[4, 3] => 7 
[4, 4] => 8 
